### PR TITLE
fix: use correct LXD profile on state.Charm struct

### PR DESCRIPTION
**Pull request by wallyworld**
The terraform smoke tests fail intermittently deploying a charm. The error is:

 "charm does no implement LXDProfiler"

It turns out that with async charm download we can pass to insertCharmOps() a state.Charm struct. This struct has the method

`func (c *Charm) LXDProfile() *LXDProfile`

where it needs to be

`func (c *Charm) LXDProfile() *charm.LXDProfile`

This PR fixes the method signature so that state.Charm now implements charm.LXDProfiler.

As a drive by, use xxlarge runner for terraform smoke test job.

## QA steps

bootstrap 
juju deploy --revision=1 --channel=stable --base ubuntu@20.04 juju-qa-refresher
juju refresh refresher

## Links

**Jira card:** [JUJU-6326](https://warthogs.atlassian.net/browse/JUJU-6326)

[JUJU-6326]: https://warthogs.atlassian.net/browse/JUJU-6326?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: state/charm.go [hunk 1]
```diff
@@ -251,7 +251,7 @@ func insertCharmOps(mb modelBackend, info CharmInfo) ([]txn.Op, error) {
 	}
 	lpc, ok := info.Charm.(charm.LXDProfiler)
```
Comment (hpidcock) [inline state/charm.go:252]: Why didn't a test get added that exercises this?

**Review by hpidcock — APPROVED** [review]

File: state/charm.go [hunk 2]
```diff
@@ -678,8 +678,15 @@ func (c *Charm) Actions() *charm.Actions {
 }
 
 // LXDProfile returns the lxd profile definition of the charm.
-func (c *Charm) LXDProfile() *LXDProfile {
-	return c.doc.LXDProfile
+func (c *Charm) LXDProfile() *charm.LXDProfile {
```
Comment (SimonRichardson) [inline state/charm.go:681]: Could this be a source of a lot of failures for LXD profiles?

File: state/charm.go (hunk 1 above)
Comment (wallyworld) [reply state/charm.go:252]: Added a compile time check

**Pull request merged**